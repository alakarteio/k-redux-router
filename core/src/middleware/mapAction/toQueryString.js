// https://stackoverflow.com/questions/1714786/query-string-encoding-of-a-javascript-object
export default obj => Object.keys(obj).map(k => `${encodeURIComponent(k)}=${encodeURIComponent(obj[k])}`).join('&')
