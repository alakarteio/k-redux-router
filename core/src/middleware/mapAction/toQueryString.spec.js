/* eslint-env jest */
import toQueryString from './toQueryString'

describe('toQueryString', () => {
  const match = res => () => expect(res).toMatchSnapshot()
  it('should returns the query string', match(toQueryString({ object: 'with', some: 'values' })))
})
