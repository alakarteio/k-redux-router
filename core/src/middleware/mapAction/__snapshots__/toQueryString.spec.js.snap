// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toQueryString should returns the query string 1`] = `"object=with&some=values"`;
